arch/risc-v/src/common/riscv_internal.h

```diff
@@ -121,6 +150,37 @@
 #define getreg64(a)         (*(volatile uint64_t *)(a))
 #define putreg64(v,a)       (*(volatile uint64_t *)(a) = (v))
 
+#define READ_CSR(reg) \
+  ({ \
+     uintptr_t reg##_val; \
```

Comment:
@pussuw build fixed here by `reg##_val`. Pleas review. I changed from `long` to `uintptr_t` and expect that it should be fine,

Reply:
Yes this should produce an unique enough tmp variable. I have no idea why the compiler has not complained about this before. Seems like enough obfuscation in pre-processing will silence it (I have seen similar behavior before).

Using uintptr_t here is fine, as all actual CSRs are XLEN

Reply:
If needed we can go even more unique with
```
#define __tmpname(name) tmp##name
#define __uname(name)   __tmpname(name)

uintptr_t __uname(__LINE__);
```